typescript/db/src/dev/apollo-server.ts

```diff
@@ -22,8 +21,22 @@ export const schemaWithResolvers = addResolversToSchema({
 
 const server = new ApolloServer({
   introspection: true,
-  context: () => {
-    return { repository, user: { id: v4() } };
+  context: async ({ req }) => {
+    // Get user ID from session token stored in cookie
+    const cookie = req?.headers?.cookie;
+    const parsedCookie =
+      cookie != null
+        ? Object.fromEntries(
+            cookie.split(/; */).map((c) => {
+              const [key, ...v] = c.split("=");
+              return [key, decodeURIComponent(v.join("="))];
+            })
+          )
+        : {};
+    const session = await prisma.session.findUnique({
+      where: { sessionToken: parsedCookie?.["next-auth.session-token"] },
+    });
```

Comment:
Isn't it what getSession does? 
https://next-auth.js.org/getting-started/client#getsession

Reply:
I agree, for everything related to auth, we should use next-auth hooks.

Reply:
Yes but `getSession` would not work, probably because we are in a standalone server without NextJS 🤷

Reply:
https://next-auth.js.org/getting-started/client#getsession

Did you try `const session = await getSession({ req })` ? I'm not sure if the req object is specific to NextJS

Reply:
Ok I think that now I remember why it would not work, from the docs it says:

>It calls /api/auth/session and returns a promise with a session object, or null if no session exists

Here we do not have any endpoint at that route, so it would not return any session.

Reply:
Indeed, and I didnt have this in mind because it's new in Next Auth v4. Ok then I think your solution is good